etf/evaluation/views.py

```
@@ -186,3 +191,58 @@ def view(self, request, url_data):
 FormPage(title="Other ethical", field_names=("other_ethical_information",))
 
 SimplePage(title="End")
+
+
+class EvaluationSearchForm(forms.Form):
+    id = forms.UUIDField(required=False)
+    title = forms.CharField(max_length=100, required=False)
+    description = forms.CharField(max_length=100, required=False)
+    topics = forms.MultipleChoiceField(choices=models.Topic.choices, required=False)
+    organisations = forms.MultipleChoiceField(choices=models.Organisation.choices, required=False)
+    is_published = forms.BooleanField(required=False)
+    search_phrase = forms.CharField(max_length=100, required=True)
+
+
+def search_evaluations_view(request):
+    qs = models.Evaluation.objects.all()
+    data = {}
+    errors = {}
+    if request.method == "GET":
+        form = EvaluationSearchForm(request.GET)
+        if form.is_valid():
+            id = form.cleaned_data["id"]
+            topics = form.cleaned_data["topics"]
+            organisations = form.cleaned_data["organisations"]
+            is_published = form.cleaned_data["is_published"]
+            search_phrase = form.cleaned_data["search_phrase"]
+            if id:
+                qs = qs.filter(id=id)
+            if organisations:
+                qs = qs.filter(organisation__in=organisations)
+            if is_published:
+                qs = qs.filter(is_published=True)
+            if topics:
+                topics_qs = models.Evaluation.objects.none()
+                for topic in topics:
+                    topic_qs = qs.filter(topics__contains=topic)
+                    topics_qs = topics_qs | topic_qs
+                qs = topics_qs
+            if search_phrase:
+                # TODO - what fields do we care about?
+                most_important_fields = ["title", "description", "topics", "organisation"]
+                all_fields = {f.name for f in models.Evaluation._meta.get_fields()}
+                other_fields = all_fields.difference(set(most_important_fields))
```

This is over cautious.  If we change the field names we want this to blow up, not silently fail

---

My thinking was that it would just search all the fields and I didn't want to write them all out! But I will change as we really haven't figured out what we want from the search yet anyway.